### PR TITLE
implement ILogger<>

### PR DESCRIPTION
**Pull request by jennyf19**
https://github.com/AzureAD/microsoft-identity-web/issues/3

### CODE REVIEW

File: src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs [hunk 1]
```diff
@@ -2,43 +2,51 @@
...
+        private Func<JwtBearerChallengeContext, Task> s_onChallenge;
 
         /// <summary>
         /// Subscribes to all the JwtBearer events, to help debugging, while
         /// preserving the previous handlers (which are called)
         /// </summary>
         /// <param name="events">Events to subscribe to</param>
-        public static JwtBearerEvents Subscribe(JwtBearerEvents events)
+        public JwtBearerEvents Subscribe(
+            JwtBearerEvents events)
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs]: you might want to remove the extra line?

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 2]
```diff
@@ -135,8 +138,7 @@ public class TokenAcquisition : ITokenAcquisition
             }
             catch (MsalException ex)
             {
-                // brentsch - todo, write to a log
-                Debug.WriteLine(ex.Message);
+                _logger.LogInformation(ex.Message);
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs]: LogInformation? or LogDebug?

Comment (jennyf19) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs]: @jmprieur i'm not sure. See [#8 here](https://nblumhardt.com/2017/07/library-logging/) this is why i put the others in Debug, but this is an exception.

File: src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs]: yes. thanks.

**Comment by jmprieur** [comment]
yay!

File: src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs [hunk 3]
```diff
@@ -2,43 +2,50 @@
...
+        public JwtBearerMiddlewareDiagnostics(ILogger<JwtBearerMiddlewareDiagnostics> logger)
+        {
+            _logger = logger;
+        }
+
         /// <summary>
         /// Invoked if exceptions are thrown during request processing. The exceptions will be re-thrown after this event unless suppressed.
         /// </summary>
-        private static Func<AuthenticationFailedContext, Task> s_onAuthenticationFailed;
+        private Func<AuthenticationFailedContext, Task> s_onAuthenticationFailed;
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs]: Proposal to not use 's_' prefix for statics. Also these should use proper casing.

File: src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -209,7 +210,9 @@ public static class WebApiServiceCollectionExtensions
 
                 if (subscribeToJwtBearerMiddlewareDiagnosticsEvents)
                 {
-                    options.Events = JwtBearerMiddlewareDiagnostics.Subscribe(options.Events);
+                    var diags = builder.Services.BuildServiceProvider().GetRequiredService<IJwtBearerMiddlewareDiagnostics>();
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: I would spell the name out to make it more readable.

File: src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs (hunk 3 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs]: that's something to bring up to the team. we have this as a rule in MSAL, so all the statics should shave this format.

**Pull request merged**